### PR TITLE
Finalise the options

### PR DESCRIPTION
**Pull request by sbrunner**
Make desktop and desktop_alt more identical
Make mobile and mobile_alt more identical
Add missing gmfShareOptions option
Make the showInfobar configurable
Add documentation
...

Consider reviewing commit by commit :-)

### CODE REVIEW

File: docs/guidelines.md [hunk 1]
```diff
@@ -142,6 +131,36 @@ const myComponent = {
...
+All the options of a component that can be represented as JSON should be injected throw one object in the component
+and should start with the prefix `ngeo` or `gmf` and should be describe in one of the `options.js` files with a
+type name who is the same as the injected name.
+
+The styles should use the type
+[StyleLike](https://camptocamp.github.io/ngeo/master/jsdoc/module-contribs_gmf_src_options.html#~StyleLike)
+and be build with the function
+[buildStyle](https://camptocamp.github.io/ngeo/master/jsdoc/module-contribs_gmf_src_options.html#.buildStyle).
+
+For this kind of options we shouldn't porvide default values and we use the standard injection mechanisme, then
```
Comment (ger-benjamin) [inline docs/guidelines.md:154]:
```suggestion
For this kind of options we don't have to provide  a default value. It uses the standard injection mechanism, then
```

File: docs/guidelines.md [hunk 2]
```diff
@@ -142,6 +131,36 @@ const myComponent = {
...
+the application should provide a value normally throw the `dynamic.json`.
```
Comment (ger-benjamin) [inline docs/guidelines.md:155]:
```suggestion
the application should provide a value normally through the `dynamic.json`.
```

File: docs/guidelines.md [hunk 3]
```diff
@@ -142,6 +131,36 @@ const myComponent = {
...
+
+But for the value in the option the component shouldn't fail if the argument is missing.
```
Comment (ger-benjamin) [inline docs/guidelines.md:157]:
```suggestion
For the value in the option the component don't fail if an argument is missing.
```

File: docs/guidelines.md [hunk 4]
```diff
@@ -142,6 +131,36 @@ const myComponent = {
...
+
+### Options who is a function
+
+There is no peconisation for this kind of options.
```
Comment (ger-benjamin) [inline docs/guidelines.md:161]:
```suggestion
There is no recommendation for this kind of options.
```

File: contribs/gmf/src/controllers/AbstractAppController.js [hunk 5]
```diff
@@ -524,31 +438,7 @@ export function AbstractAppController($scope, $injector, mobile) {
...
-
-  const printPanelActivate = new ngeoMiscToolActivate(this, 'printPanelActive');
-  ngeoToolActivateMgr.registerTool(mapTools, printPanelActivate, false);
-
-  const contextdataActivate = new ngeoMiscToolActivate(this, 'contextdataActive');
-  ngeoToolActivateMgr.registerTool(mapTools, contextdataActivate, false);
-
-  const routingPanelActive = new ngeoMiscToolActivate(this, 'routingPanelActive');
-  ngeoToolActivateMgr.registerTool(mapTools, routingPanelActive, false);
+  ngeoToolActivateMgr.registerTool('mapTools', queryToolActivate, true);
```
Comment (ger-benjamin) [inline contribs/gmf/src/controllers/AbstractAppController.js:437]:
```suggestion
  ngeoToolActivateMgr.registerTool(this.mapToolsGroup, queryToolActivate, true);
```
?

**Review by ger-benjamin — APPROVED** [review]
Looks working well, thanks

File: contribs/gmf/apps/mobile_alt/index.html.ejs [hunk 6]
```diff
@@ -53,8 +55,8 @@
         ng-click="mainCtrl.toggleLeftNavVisibility()">
         <%=require('gmf/icons/layers.svg')%>
       </button>
-      <gmf-search gmf-search-map="mainCtrl.map"
-        gmf-search-datasources="mainCtrl.searchDatasources"
+      <gmf-search gmf-search-map="::mainCtrl.map"
+        gmf-search-datasources="::mainCtrl.searchDatasources"
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply contribs/gmf/apps/mobile_alt/index.html.ejs]: It was removed in a layer commit :-)

File: contribs/gmf/examples/print.html [hunk 7]
```diff
@@ -46,7 +46,6 @@
             gmf-print-map="::ctrl.map"
             gmf-print-active="printActive"
             gmf-print-fieldvalues="::ctrl.defaulPrintFieldstValues"
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply contribs/gmf/examples/print.html:48]: +1 thanks :-)

File: docs/guidelines.md [hunk 8]
```diff
@@ -142,6 +131,36 @@ const myComponent = {
 };
 ```
 
+
+## Options
+
+We have 3 types of configuragle input/output of an Angular component.
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply docs/guidelines.md]: Thanks :-)

File: docs/guidelines.md [hunk 9]
```diff
@@ -142,6 +131,36 @@ const myComponent = {
...
+
+### Link between component
+
+The variable that used to share data between component should be provided throw the HTML attributes.
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply docs/guidelines.md]: +1

File: docs/guidelines.md [hunk 10]
```diff
@@ -142,6 +131,36 @@ const myComponent = {
...
+
+### Options that can be represent ad JSON
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply docs/guidelines.md]: Done :-)

File: docs/guidelines.md [hunk 11]
```diff
@@ -142,6 +131,36 @@ const myComponent = {
...
+
+All the options of a component that can be represented as JSON should be injected throw one object in the component
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply docs/guidelines.md]: +1

File: docs/guidelines.md [hunk 12]
```diff
@@ -142,6 +131,36 @@ const myComponent = {
...
+and should start with the prefix `ngeo` or `gmf` and should be describe in one of the `options.js` files with a
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply docs/guidelines.md]: +1

File: docs/guidelines.md (hunk 1 above)
Comment (sbrunner) [reply docs/guidelines.md]: +1

File: docs/guidelines.md (hunk 3 above)
Comment (sbrunner) [reply docs/guidelines.md]: +1

File: docs/guidelines.md (hunk 4 above)
Comment (sbrunner) [reply docs/guidelines.md]: +1

File: contribs/gmf/src/controllers/AbstractAppController.js (hunk 5 above)
Comment (sbrunner) [reply contribs/gmf/src/controllers/AbstractAppController.js]: Looks good :-)

**Pull request merged**